MultipleFileUpload/Model/NetteDatabase/Queues.php

```diff
@@ -48,9 +48,17 @@ public function getFilesTable(){
 		return self::$filesTable;
 	}
 	
-	public function __construct(\Nette\Database\Context $database)
+	public function __construct($database, $tempDir)
 	{
-		$this->database = $database;
+		// TODO: wrong; there must be not dependency on container here
+		$connection = new Nette\Database\Connection($database['dsn'], $database['user'], $database['password']);
+		self::$uploadsTempDir = $tempDir . DIRECTORY_SEPARATOR . "uploads-MFU";
+		if(!file_exists(self::$uploadsTempDir)) {
+			mkdir(self::$uploadsTempDir, 0775, TRUE);
+		};
+		$cacheStorage = new Nette\Caching\Storages\FileStorage(self::$uploadsTempDir);
+		$structure = new Nette\Database\Structure($connection, $cacheStorage);
+		$this->database = new Nette\Database\Context($connection, $structure);
```

Comment:
Whole Database\Context should be injected. IT is responsibility of DI container to create connection.